contracts/MarginCalculator.sol

```diff
@@ -345,6 +368,128 @@ contract MarginCalculator is Ownable {
         return cashValueInCollateral.toScaledUint(collateralDecimals, true);
     }
 
+    // structs to avoid stack too deep error
+    // struct to store shortAmount, shortStrike and shortUnderlyingPrice scaled to 1e27
+    struct ShortScaledDetails {
+        FPI.FixedPointInt shortAmount;
+        FPI.FixedPointInt shortStrike;
+        FPI.FixedPointInt shortUnderlyingPrice;
+    }
+    // struct to store isLiquidatable and the price for 1 otoken repaid scaled to the collateral decimals
+    struct LiquidationStatus {
+        bool isLiquidatable;
+        uint256 debtPrice;
+    }
+
+    /**
+     * @notice check if a specific vault is undercollateralized at a specific chainlink round
+     */
+    function isLiquidatable(
+        MarginVault.Vault memory _vault,
+        uint256 _vaultType,
+        uint256 _vaultLatestUpdate,
+        uint256 _roundId
+    )
+        external
+        view
+        returns (
+            bool,
+            uint256,
+            uint256
+        )
+    {
+        // create LiquidationStatus struct to avoid stack too deep error
+        // will store isLiquidatable and the liquidation price
+        LiquidationStatus memory liquidationStatus = LiquidationStatus(false, 0);
+
+        // liquidation is only supported for naked margin vault
+        require(_vaultType == 1, "MarginCalculator: invalid vault type to liquidate");
+
+        // get vault details
+        VaultDetails memory vaultDetails = _getVaultDetails(_vault, _vaultType);
+
+        // can not liquidate vault that have no short position
+        if (!vaultDetails.hasShort) return (false, 0, 0);
+
+        // do not allow liquidating expired position
+        require(
+            now < vaultDetails.shortExpiryTimestamp,
+            "MarginCalculator: can not liquidate vault with expired short otoken"
+        );
+
+        // get otoken underlying asset price and timestamp from chainlink feed
+        (uint256 price, uint256 timestamp) = oracle.getChainlinkRoundData(
+            vaultDetails.shortUnderlyingAsset,
+            uint80(_roundId)
+        );
+
+        // check that price timestamp is after latest timestamp the vault was updated at
+        require(
+            timestamp >= _vaultLatestUpdate,
+            "MarginCalculator: auction timestamp should be post vault latest update"
+        );
```

Comment:
again, why not just return `false`? 
I think it's better to avoid using revert in a view functions, so it's easier to other contracts to integrate with. 
It's kind of inconsistent calling a function will sometime cause reverts and sometime return false.